internal/nomad/sentry_debug_writer.go

```diff
@@ -0,0 +1,89 @@
+package nomad
+
+import (
+	"context"
+	"fmt"
+	"github.com/getsentry/sentry-go"
+	"io"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+const (
+	// timeDebugMessageFormat is the format of messages that will be converted to debug messages.
+	timeDebugMessageFormat = "echo -ne \"\\x1EPoseidon %s $(date +%%s%%3N)\\x1E\""
+	// Format Parameters: 1. Debug Comment, 2. command.
+	timeDebugMessageFormatStart = timeDebugMessageFormat + "; %s"
+	// Format Parameters: 1. command, 2. Debug Comment.
+	timeDebugMessageFormatEnd = "%s && " + timeDebugMessageFormat
+
+	timeDebugMessagePatternGroupText = 1
+	timeDebugMessagePatternGroupTime = 2
+)
+
+var (
+	timeDebugMessagePattern      = regexp.MustCompile(`\x1EPoseidon (?P<text>\w+) (?P<time>\d+)\x1E`)
+	timeDebugMessagePatternStart = regexp.MustCompile(`\x1EPoseidon`)
+)
+
+// SentryDebugWriter is scanning the input for the debug message pattern.
+// For matches, it creates a Sentry Span. Otherwise, the data will be forwarded to the Target.
+// The passed context Ctx should contain the Sentry data.
+type SentryDebugWriter struct {
+	Target   io.Writer
+	Ctx      context.Context
+	lastSpan *sentry.Span
+}
+
+// Improve: Handling of a split debug messages (usually p is exactly one debug message, not less and not more).
+func (s *SentryDebugWriter) Write(p []byte) (n int, err error) {
+	if !timeDebugMessagePatternStart.Match(p) {
+		count, err := s.Target.Write(p)
+		if err != nil {
+			err = fmt.Errorf("SentryDebugWriter Forwarded: %w", err)
```

Comment:
True. Since the error is just forwarded and the actual error will come from the http writer, `codeOceanOutputWriter`, `Ls2JsonWriter` or `ContentLengthWriter`, I wanted to keep it short. Wdyt about this slight adjustment or what would you suggest?

```suggestion
			err = fmt.Errorf("SentryDebugWriter Forwarded Error: %w", err)
```